hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/virtual/schedule/ScheduleEqualityVirtualValue.java

```diff
@@ -95,63 +99,75 @@ public String toString() {
+        size += Integer.BYTES; // ids size
+        for (var e : ids.entrySet()) {
+            size += Integer.BYTES + e.getKey().getBytes(StandardCharsets.UTF_8).length + Long.BYTES;
         }
-        number = in.readLong();
+        size += Long.BYTES;
+        return size;
     }
 
-    @Override
-    public void deserialize(ByteBuffer in, int version) throws IOException {
-        int s = in.getInt();
-        ids.clear();
-        for (int x = 0; x < s; ++x) {
-            byte[] keyBytes = new byte[in.getInt()];
-            in.get(keyBytes);
-            var k = new String(keyBytes, StandardCharsets.UTF_8);
-            ids.put(k, in.getLong());
+    private <E extends Exception> void serializeTo(
+            final CheckedConsumerE<Integer, E> writeIntFn,
+            final CheckedConsumerE<Long, E> writeLongFn,
+            final CheckedConsumerE<byte[], E> writeBytesFn)
+            throws E {
+        writeIntFn.accept(ids.size());
+        for (var e : ids.entrySet()) {
+            var keyBytes = e.getKey().getBytes(StandardCharsets.UTF_8);
+            writeIntFn.accept(keyBytes.length);
+            writeBytesFn.accept(keyBytes);
+            writeLongFn.accept(e.getValue());
         }
-        number = in.getLong();
+        writeLongFn.accept(number);
     }
 
     @Override
-    public void serialize(SerializableDataOutputStream out) throws IOException {
-        out.writeInt(ids.size());
-        for (var e : ids.entrySet()) {
-            var keyBytes = e.getKey().getBytes(StandardCharsets.UTF_8);
-            out.writeInt(keyBytes.length);
-            out.write(keyBytes);
-            out.writeLong(e.getValue());
-        }
-        out.writeLong(number);
+    public void serialize(final SerializableDataOutputStream out) throws IOException {
+        serializeTo(out::writeInt, out::writeLong, out::write);
+    }
+
+    void serialize(final WritableSequentialData out) {
+        serializeTo(out::writeInt, out::writeLong, out::writeBytes);
     }
 
     @Override
-    public void serialize(ByteBuffer out) throws IOException {
-        serializeReturningBytesWritten(out);
+    public void serialize(final ByteBuffer buffer) throws IOException {
+        serializeTo(buffer::putInt, buffer::putLong, buffer::put);
     }
 
-    int serializeReturningBytesWritten(ByteBuffer out) {
-        int bytesWritten = 0;
-        out.putInt(ids.size());
-        bytesWritten += Integer.BYTES;
-        for (var e : ids.entrySet()) {
-            var keyBytes = e.getKey().getBytes(StandardCharsets.UTF_8);
-            out.putInt(keyBytes.length);
-            out.put(keyBytes);
-            out.putLong(e.getValue());
-            bytesWritten += Integer.BYTES + keyBytes.length + Long.BYTES;
+    private <E extends Exception> void deserializeFrom(
+            final CheckedSupplierE<Integer, E> readIntFn,
+            final CheckedSupplierE<Long, E> readLongFn,
+            final CheckedConsumerE<byte[], E> readBytesFn)
```

Comment:
I mostly kept existing code, which is something like

```java
    final byte[] someField = new byte[len];
    readBytesFn.accept(someField);
```

It can be rewritten to use a supplier rather than a consumer, and that supplier will allocate the array, but I won't be able to use a method handle then. Not a big deal, but not very nice...